scripts/us_census/pep_srh/process.py

```diff
@@ -0,0 +1,1057 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+'''
+This module creates CSV files used for importing data into DC.
+
+Below are list of files processed -
+County
+    1990 - 2000     Processed As Is
+    2000 - 2010     Processed As Is
+    2010 - 2020     Processed As Is
+
+State
+    1980 - 1990     Processed As Is
+    1990 - 2000     Processed As Is
+    2000 - 2010     Aggregated from County to state
+                    (data matches with State Level files available)
+    2010 - 2020     Aggregated from County to state
+                    (data matches with State Level files available)
+
+National
+    1980 - 1990     Aggregted from State to National Level
+    1990 - 2000     Aggregted from State to National Level
+    2000 - 2010     Aggregted from State to National Level
+    2010 - 2020     Aggregted from State to National Level
+
+Also SV aggregation are produced while processing above files.
+E.g., Count_Person_White_HispanicOrLatino is calulated by addding
+Count_Person_Male_WhiteHispanicOrLatino and
+Count_Person_Male_WhiteHispanicOrLatino
+
+Before running this module, run download.sh script, it downloads required
+input files, created necessary folders for processing.
+
+Folder information
+download_files - downloaded files are placed here
+process_files - intermediate processed files are placed in this folder.
+output_files - output files (mcf, tmcf and csv are written here)
+'''
+
+import os
+import pandas as pd
+
+from constants import DOWNLOAD_DIR, PROCESS_AS_IS_DIR, PROCESS_AGG_DIR
+from constants import OUTPUT_DIR, stat_var_col_mapping, working_directories
+
+_CODEDIR = os.path.dirname(os.path.realpath(__file__))
+
+
+def process_national_files_1980_1990():
+    '''
+    Process for National Level data, 1980 - 1990
+    Uses State Level file as input
+    '''
+    # Section 1 - Writing As Is Data
+    input_file_path = _CODEDIR + PROCESS_AS_IS_DIR + '1980_1990/state/'
+    output_file_path = _CODEDIR + PROCESS_AS_IS_DIR + '1980_1990/national/'
+    output_file_name = 'national_1980_1990.csv'
+    files_list = os.listdir(input_file_path)
+    files_list.sort()
+
+    df = pd.read_csv(input_file_path + files_list[0])
+    df['LOCATION'] = 'country/USA'
+    df = df.groupby(['YEAR', 'LOCATION']).agg('sum').reset_index()
+    df.to_csv(output_file_path + output_file_name, index=False)
+
+    # Section 2 - Writing Agg data
+    input_file_path = _CODEDIR + PROCESS_AGG_DIR + '1980_1990/state/'
+    output_file_path = _CODEDIR + PROCESS_AGG_DIR + '1980_1990/national/'
+    files_list = os.listdir(input_file_path)
+    files_list.sort()
+
+    df = pd.read_csv(input_file_path + files_list[0])
+    df['LOCATION'] = 'country/USA'
+    df = df.groupby(['YEAR', 'LOCATION']).agg('sum').reset_index()
+    df.to_csv(output_file_path + output_file_name, index=False)
+
+
+def process_national_files_1990_2000():
```

Comment:
process_national_files_1980_1990(), process_national_files_1990_2000(), process_national_files_2010_2020()
all are same except for the hardcoded folder path '1990_2000'. 
Pls make that an argument and combine them into a single function that can be invoked multiple times fro each path.